### ISSUE TITLE
Issues with container build

### ISSUE THREAD

**Issue by mbach04** [CMYanko/ubi-nexus3#2]
Hi Curtis,

A couple of issues I've observed when duplicating this in our environment.

- [ ] It calls out to the internet heavily during the build process. This isn't a problem per se, rather it's a difficulty we have to resolve to enable hosting of the product from within the DSOP. The build process must be repeated completely offline. Is there an easy way to override this behaviour? [see output msg]
- [ ] I don't believe sufficient ports are being exposed to enable the docker registry. In other versions of Nexus I've seen this ride across port 18081. 

### output msg
```Getting information for chef stable  for el...
downloading https://www.getchef.com/stable/chef/metadata?v=&p=el&pv=7&m=x86_64
  to file /tmp/install.sh.9/metadata.txt
trying curl...
sha1	bcb835748819813842b664d7bdb1cf509eb71c09
sha256	85365d0a4c68a3445743d18e70055d762648bf673fc95f1e0e17194701f64d5f
url	https://packages.chef.io/files/stable/chef/15.0.300/el/7/chef-15.0.300-1.el7.x86_64.rpm
version	15.0.300
downloaded metadata file looks valid...
downloading https://packages.chef.io/files/stable/chef/15.0.300/el/7/chef-15.0.300-1.el7.x86_64.rpm
```

Some assistance on these items would be much appreciated.

Thanks!

**Comment by CMYanko** [comment]
Correct, it does call out for both chef and the nexus binary. For an offline environment you’ll need to pre-populate some corresponding local cache I suppose.

You are correct about the ports but then no assumption is made about the eventual configuration. I do have an example in my reference platform project that creates a separate blog store for docker and then creates the proxy and hosted registries along with port assignments which are  aligned to reverse proxy for SSL offloading. Essentially, it all depends on how the end state will look. Opening more ports is trivial and can be added in your implementation of this dockerfile.

**Comment by CMYanko** [comment]
I also see those chef packages rpm’s, are they in the red hat registry you’ll be installing package from?

**Comment by mbach04** [comment]
Yes, adding an EXPOSE is trivial, but I was guessing there was more configuration on the backend to get the whole thing working as you described above. Those rpm's aren't available in the UBI channels, unfortunately.

**Comment by mbach04** [comment]
Curtis, would it be possible to get a Dockerfile(s) and helm chart from you that would comprise a full Nexus deployment including a docker registry?

I took a look at the platform reference repo you mentioned, but it wasn't clear what parts would be required to create a solution.

**Comment by CMYanko** [comment]
Sadly we don't have any helm charts that we've created. There is a config script in the reference platform to add docker and I'll point that out in our email thread
